### ISSUE TITLE
Clarify style guideline for constants

### ISSUE THREAD

**Issue by vansimke** [dojo/meta#171]
The current [style](https://github.com/dojo/meta/blob/master/STYLE.md) guide states that constants should always be named with uppercase letters separated by underscores. However, the current accepted practice appears to group constants into two groups. 

One group holds symbols that are used to clarify the intent of the code (for example, replacing magic strings with a name its meaning - `const NOT_FOUND_STATUS = 404`. This use of constants is in compliance with the current style guide. 

The second use of constants is to define "variables" whose values never be reassigned. An example of this is when local variables are extracted from a widgets properties object using a destructuring assignment like this:
```ts
const {
	firstName = 'firstName',
	lastName = 'lastName'
} = this.properties;
```

**Comment by rishson** [comment]
My current opinion is that
`const` 'variables' should be camelCased
magic number constants SHOULD_BE_CAPITALISED.

**Comment by dylans** [comment]
Fixed in #220
